src/common/writefile.cpp

```diff
@@ -94,8 +94,9 @@ size_t WriteFile::write(const void *dataPtr, size_t dataSize) {
 
 	assert(dataPtr);
 
+	const size_t oldPos = pos();
 	const size_t written = std::fwrite(dataPtr, 1, dataSize, _handle);
-	_size += written;
+	_size += MAX(static_cast<size_t>(0), -size() + oldPos + written);
```

Comment:
You could use `MAX<size_t>()` instead, if you want to specifiy that it should use the `size_t` version of the MAX template.